### PR TITLE
🔔 ARTESCA-8219: Notification Center 🔔 

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: Shell UI

**Context**:  We'd like to bring Notification Center to the Navbar. 

**Summary**: This Pull Request only includes the UI changes for the Notification Center. We need to address the Version, License, and Alerts aspects to enable publishing and unpublishing of Notifications.

**Acceptance criteria**: 

image

image

### CODE REVIEW

File: shell-ui/src/NotificationCenterProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  id: string;
+  title: string;
+  description: string;
+  severity: 'critical' | 'warning' | 'info';
+  redirectUrl: string;
+  createdAt: Date;
+};
+
+export type InternalNotification = Notification & {
+  readOn?: Date;
```
Comment (JBWatenbergScality) [inline shell-ui/src/NotificationCenterProvider.tsx]: nit: I think we should either use On or At suffix for the date (I'd rather prefer the On suffix)

File: shell-ui/src/NotificationCenterProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    switch (action.type) {
+      case NotificationActionType.PUBLISH:
+        const storedReadOn = localStorage.getItem(
+          LOCAL_STORAGE_NOTIFICATION_PREFIX + action.notification.id,
+        );
+        const readOn = storedReadOn ? new Date(storedReadOn) : undefined;
+        // if the Notification is already stored, direct return the state
+        if (state.find((n) => n.id === action.notification.id)) {
+          return state;
+        }
```
Comment (JBWatenbergScality) [inline shell-ui/src/NotificationCenterProvider.tsx]: I'm not sure about this approach, shall we update the fields of the old one with the newly published notification 🤔 ?

File: shell-ui/src/navbar/NotificationCenter.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                    text={`${notifications.filter((n) => !n.readOn).length}`}
+                    variant="selectedActive"
+                  />
+                )}
+              </Wrap>
+            </NotificationCenterHeader>
+            {notifications.length === 0 && (
+              <Box m={spacing.r16}>
+                <Text color="textSecondary">
+                  You have no new notifications at the moment.
```
Comment (JBWatenbergScality) [inline shell-ui/src/navbar/NotificationCenter.tsx]: nit: would be nice to add the small sleeping bell gif icon

File: shell-ui/src/NotificationCenterProvider.tsx (hunk 2 above)
Comment (ChengYanJin) [reply shell-ui/src/NotificationCenterProvider.tsx]: Indeed, should update the old one with the newly pushed one.

**Review by JBWatenbergScality — APPROVED** [review]
LGTM !!! 
phone-ringing-gif-by-xxyz

**Pull request merged**